### PR TITLE
pynndescent: n_jobs, metrics, verbose

### LINKED ISSUE #146: Pynndescent in t-SNE with k=15 seems a lot slower than in UMAP
I've just noticed that UMAP 0.4.6 with pynndescent 0.5 can be quite a bit faster to construct the kNN graph compared to openTSNE with uniform kernel and k=15. This does not make sense to me. I added verbose=1 to the pynndescent call within openTSNE and here is what I see:

UMAP - 34 sec:

```
UMAP(angular_rp_forest=True, metric='cosine', verbose=1)
Construct fuzzy simplicial set
Mon Oct 19 13:54:22 2020 Finding Nearest Neighbors
Mon Oct 19 13:54:22 2020 Building RP forest with 21 trees
Mon Oct 19 13:54:32 2020 metric NN descent for 17 iterations
	 1  /  17
	 2  /  17
	 3  /  17
	 4  /  17
	 5  /  17
	 6  /  17
	Stopping threshold met -- exiting after 6 iterations
Mon Oct 19 13:54:58 2020 Finished Nearest Neighbor Search
```

tSNE -- 61 sec

```
===> Finding 15 nearest neighbors using NN descent approximate search using cosine distance...
Mon Oct 19 14:03:15 2020 Building RP forest with 21 trees
Mon Oct 19 14:03:33 2020 metric NN descent for 17 iterations
	 1  /  17
	 2  /  17
	 3  /  17
	 4  /  17
	 5  /  17
	 6  /  17
	Stopping threshold met -- exiting after 6 iterations
   --> Time elapsed: 61.53 seconds
```

I ran both snippets twice to make sure all numba code was pre-compiled.

I don't know why this happens. The `pynndescent` call in UMAP looks identical to the one in t-SNE. https://github.com/lmcinnes/umap/blob/master/umap/umap_.py#L290

### PR DESCRIPTION
**Pull request by dkobak**
Fixes #146, #145, and updates metrics as per #142.

### CODE REVIEW

**Comment by dkobak** [comment]
I don't understand why all these tests failed :-(

**Comment by pavlin-policar** [comment]
[This test](https://github.com/pavlin-policar/openTSNE/blob/master/tests/test_nearest_neighbors.py#L279-L283) will need to be updated :)

File: openTSNE/nearest_neighbors.py [hunk 1]
```diff
@@ -398,7 +389,8 @@ def build(self, data, k):
             n_trees=n_trees,
             n_iters=n_iters,
             max_candidates=60,
-            n_jobs=n_jobs_pynndescent,
+            n_jobs=self.n_jobs,
+            verbose=1 if self.verbose>1 else 0,
```
Comment (pavlin-policar) [inline openTSNE/nearest_neighbors.py:393]: A bit nit-picky, but could you add spaces around the `>`, so `verbose=1 if self.verbose > 1 else 0`, or better yet `verobose=self.verbose > 1`.

Comment (dkobak) [reply openTSNE/nearest_neighbors.py]: Yes, that's a better way to write it, thanks. Done.

**Comment by dkobak** [comment]
Oh right. I fixed the test. I'm always confused when one of the tests fails because it's not obvious which test has failed...

**Comment by pavlin-policar** [comment]
Thanks! Also, if you scroll to the top of the [test output](https://dev.azure.com/pavlingp/openTSNE/_build/results?buildId=304&view=logs&j=9900b494-448c-5226-d8d0-3aa940fadb03&t=28aaf5cc-09e3-5d10-107a-1e82bc77f3c5), you can see exactly which tests failed. E.g. line 118.

**Pull request merged**